_episodes/07-vector-data-in-python.md

```diff
@@ -149,126 +165,221 @@ array([109222.03325 , 469461.512625, 140295.122125, 510939.997875])
 
-We can convert these coordinates to a bounding box or acquire the index of the dataframe to access the geometry. Either of these polygons can be used to clip rasters (more on that later).
 
 ## Selecting spatial features
-Sometimes, the loaded data can still be too large. We can cut it is to a even smaller extent using the `.cx` indexer (note the use of square brackets instead of round brackets, which are used instead with functions and methods):
+
+From now on, we will take in a point dataset `brogmwvolledigeset.zip`, which is the underground water monitoring wells. We will perform vector processing on this dataset, together with the cropped field polygons `fields_cropped.shp`.
+
+Let's read the two datasets.
 
 ~~~
-# Define a Boundingbox in RD
-xmin, xmax = (120_000, 135_000)
-ymin, ymax = (485_000, 500_000)
-cropfield_crop = cropfield.cx[xmin:xmax, ymin:ymax]
+fields =  gpd.read_file("data/fields_cropped.shp")
+wells = gpd.read_file("data/brogmwvolledigeset.zip")
 ~~~
 {: .language-python}
 
+And take a look at the wells:
+
+~~~
+wells.plot(markersize=0.1)
+~~~
+{: .language-python}
+![All wells](../fig/E07-03-wells-nl.png)
 
-This will cut out a smaller area, defined by a box in units of the projection, discarding the rest of the data. The resultant GeoDataframe, which includes all the features intersecting the box, is found in the `cropfield_crop` object. Note that the edge elements are not 'cropped' themselves. We can check the total bounds of this new data as before:
+The points represents all the wells over the Netherlands. Since the wells are in the lat/lon coordinates. To make it comparable with fields, we need to transfer the CRS to RD first:
 
 ~~~
-cropfield_crop.total_bounds
+wells = wells.to_crs(28992)
 ~~~
 {: .language-python}
 
+Now we would like to compare the wells with the cropped fields. We can select the wells within the fields using the `.clip` function:
+
+~~~
+wells_clip = wells.clip(fields)
+wells_clip
 ~~~
-array([119594.384, 484949.292625, 135375.77025, 500782.531])
+{: .language-python}
+
+~~~
+bro_id delivery_accountable_party quality_regime  ...
+40744  GMW000000043703  27364178    IMBRO/A    ...
+38728  GMW000000045818  27364178    IMBRO/A    ...
+...                ...       ...        ...    ...
+40174  GMW000000043963  27364178    IMBRO/A    ...
+19445  GMW000000024992  50200097    IMBRO/A    ...
+[79 rows x 40 columns]
 ~~~
 {: .output}
 
-We can then save this cropped dataset for use in future, using the `to_file()` method of our GeoDataFrame object:
+After this selection, all the wells outside fields are dropped. This takes a while to execute, because we are clipping a relatively large number of points with many polygons.
+
```

Comment:
I like this addition! But it might be easier to follow if you mention here where this is going (i.e. we first increase the size of the polygons such that they overlap and then we merge them)?

Reply:
Thx! I added some descriptions.